components/webapps/blacklist-whitelist-service/src/main/java/com/wso2telco/services/bw/Queries.java

```diff
@@ -56,360 +51,262 @@
-		}
+		appId = replaceAllWithWildcard(appId);
+		apiId = replaceAllWithWildcard(apiId);
 
-		//String apiID = blackListReq.getAPIID();
-		String userID = blackListReq.getUserID();
-		String[] msisdnList = blackListReq.getMsisdnList();
+		BlacklistWhitelistDTO blacklistWhitelistDTO = new BlacklistWhitelistDTO();
 
-		final StringBuilder errorMSG = new StringBuilder();
+		blacklistWhitelistDTO.setApiID(apiId);
+		blacklistWhitelistDTO.setAppId(appId);
+		blacklistWhitelistDTO.addMsisdn(requestDTO.getMsisdn());
+		blacklistWhitelistDTO.setAction(requestDTO.getAction());
+		blacklistWhitelistDTO.setServiceProvider(requestDTO.getSp());
+		blacklistWhitelistDTO.setUser(requestDTO.getUser());
 
-		errorMSG.append("{").append("\"Failed\":").append("{").append("\"messageId\":\"".intern())
-				.append("Blacklist Numbers".intern()).append("\",");
-		errorMSG.append("\"text\":\"").append("Blacklist Numbers could not be added to the system ".intern())
-				.append("\",");
-		errorMSG.append("\"variables\":").append(gson.toJson(msisdnList)).append("}}");
-
-		final StringBuilder successMSG = new StringBuilder();
-		successMSG.append("{").append("\"Success\":").append("{").append("\"messageId\":\"").append("Blacklist Numbers")
-				.append("\",");
-		successMSG.append("\"text\":\"").append("Blacklist Numbers Successfully Added to the system ").append("\",");
-		successMSG.append("\"variables\":").append(gson.toJson(msisdnList)).append("}}");
-
-		//gson = new Gson();
-		String[] apiInfoArray = blackListWhiteListService.getAPIInfo(apiId);
-		if (userID != null && msisdnList != null && apiInfoArray.length != 0 && apiInfoArray.length == 3) {
-			try {
-				BlackListDTO blackListDTO = new BlackListDTO();
-				blackListDTO.setApiID(String.valueOf(apiId));
-				blackListDTO.setApiName(apiInfoArray[0] + ":" + apiInfoArray[1] + ":" + apiInfoArray[2]);
-				blackListDTO.setUserID(userID);
-				blackListDTO.setUserMSISDN(msisdnList);
-				blackListDTO.setValidationRegex(blackListReq.getValidationRegex());
-				blackListDTO.setValidationPrefixGroup(blackListReq.getValidationPrefixGroup());
-				blackListDTO.setValidationDigitsGroup(blackListReq.getValidationDigitsGroup());
-
-
-				blackListWhiteListService.blacklist(blackListDTO);
-
-				return Response.status(Response.Status.OK).entity(successMSG.toString()).build();
+		try {
+			Response.Status status = Response.Status.CREATED;
+			APIBlacklistWhitelistResponseDTO responseDTO = blackListWhiteListService.blacklist(blacklistWhitelistDTO);
 
-			} catch (BusinessException ex) {
-				StringBuilder errorMessage = new StringBuilder("{ \"Failed\": { \"messageId\": \"Blacklist Numbers\", \"text\": \"Blacklist Numbers could not be added to the system\", \"variables\": \"" + ex.getErrorType().toString() + "\" } }");
-				return Response.status(Response.Status.BAD_REQUEST).entity(errorMessage.toString()).build();
+			if(responseDTO.getFailed() > 0){
```

Comment:
You might need a better logic here. Although responseDTO contains failed values, the service will insert all the valid values to the database from the blackListWhiteListService.blacklist(blacklistWhitelistDTO); call. So returning a bad request to the user is misleading.

Reply:
this was a requirement from ui end